### PR TITLE
Feature/next lap invalidation

### PR DESCRIPTION
**Pull request by jamburger71**
New code changes to allow for next lap invalidation from CC blocks, using ROBLOX attribute values.

Requires the starter CC block to be updated with the Boolean "InvalidsNextLap" attribute to function.

### CODE REVIEW

File: src/StarterGui/WesTimingGui/TimingLocalScript/PlayerBehaviourHandler.lua [hunk 1]
```diff
@@ -24,6 +24,11 @@ function handler.HandlePlayerTouch(part: BasePart)
     elseif part.Name == "_S2" then
         _keeperService:HandleSector(2)
     elseif part.Name == "_CC" then
+        -- Checks if CC invalidates next lap
+        local nextLapInvalid = part:GetAttribute("InvalidsNextLap")
```
Comment (Wesley1041) [inline src/StarterGui/WesTimingGui/TimingLocalScript/PlayerBehaviourHandler.lua:28]: Can you expand `README.lua` with an explanation on how to do this? This feature should be known from the readme.

**Review by Wesley1041 — CHANGES_REQUESTED** [review]
Looks good, but needs a few tweaks before it can be approved. The main thing is that the new feature should be mentioned in the readme.

Can you also update the version number in the readme to 1.1? In the future I think there should be a develop branch.

File: src/StarterGui/WesTimingGui/TimingLocalScript/TimingKeeperService.lua [hunk 2]
```diff
@@ -65,7 +66,12 @@ function service:HandleLap()
 
     -- Reset states
     currentSector = 1
-    lapIsValid = true
+    if nextLapIsValid then
+        lapIsValid = true
+        _popupService:NewPopup("CURRENT LAP INVALIDATED BY PREVIOUS LAP CORNER CUT", Color3.fromRGB(170, 41, 41), 300)
```
[the remark this answers is not in the document]
Comment (jamburger71) [reply src/StarterGui/WesTimingGui/TimingLocalScript/TimingKeeperService.lua:71]: I copied the colour from the already existing popup for CCs - should I change that one too?

File: src/StarterGui/WesTimingGui/TimingLocalScript/PlayerBehaviourHandler.lua [hunk 3]
```diff
@@ -24,6 +24,11 @@ function handler.HandlePlayerTouch(part: BasePart)
...
+        if nextLapInvalid then
+            _keeperService:AddNextLapCut()
```
Comment (Wesley1041) [inline src/StarterGui/WesTimingGui/TimingLocalScript/PlayerBehaviourHandler.lua:30]: This is going to trigger the prompt twice. Once for a regular CC and one for the next lap CC

File: src/StarterGui/WesTimingGui/TimingLocalScript/TimingKeeperService.lua [hunk 4]
```diff
@@ -86,6 +92,14 @@ function service:AddCornerCut()
     _remoteHandler:RequestAddCornerCut()
 
 end
+-- Registers next lap as invalid if a player goes through certain CCs with attributes
+function service:AddNextLapCut()
+    
+    if nextLapIsValid then
+        _popupService:NewPopup("NEXT LAP INVALIDATED", Color3.fromRGB(170, 41, 41), 300)
```
Comment (Wesley1041) [inline src/StarterGui/WesTimingGui/TimingLocalScript/TimingKeeperService.lua:99]: Hardcoded Color3 again

**Comment by Wesley1041** [comment]
Line 24 of PopupService.lua is triggering the following error: 
`Unable to assign property BackgroundColor3. Color3 expected, got nil`

This happens when you cross the line after you have a supposed invalid lap. The lap still gets added as new best lap, even though your last lap will be displayed in red.
image

**Review by Wesley1041 — APPROVED** [review]

**Pull request merged**